lib/modules/manager/npm/post-update/yarn.ts

```diff
@@ -90,6 +90,41 @@ export function isYarnUpdate(upgrade: Upgrade): boolean {
   return upgrade.depType === 'packageManager' && upgrade.depName === 'yarn';
 }
 
+async function determineYarnConstraint(
+  lockFileDir: string,
+  config: Partial<PostUpdateConfig<NpmManagerData>>,
+  upgrades: Upgrade[],
+): Promise<string | undefined> {
+  const yarnUpdate = upgrades.find(isYarnUpdate);
+  if (yarnUpdate?.newValue) {
+    logger.debug('Using yarn update newValue as constraint');
+    return yarnUpdate.newValue;
+  }
+  if (config.constraints?.yarn) {
+    logger.debug('Using yarn constraint from config');
+    return config.constraints.yarn;
+  }
+  const lazyPgkJson = lazyLoadPackageJson(lockFileDir);
+  const packageManagerVersion = getPackageManagerVersion(
+    'yarn',
+    await lazyPgkJson.getValue(),
+  );
+  if (packageManagerVersion) {
+    logger.debug('Using yarn constraint from package.json>packageManager');
+    return packageManagerVersion;
+  }
+  const lockFileName = upath.join(lockFileDir, 'yarn.lock');
+  const lockConstraint = getYarnVersionFromLock(
+    await getYarnLock(lockFileName),
+  );
+  if (lockConstraint) {
+    logger.debug('Using yarn constraint from yarn.lock');
+    return lockConstraint;
+  }
+  logger.debug('No yarn constraint found');
+  return undefined;
```

Comment:
These lines are unreachable due to https://github.com/renovatebot/renovate/blob/66db519d3707e606e5849e521b58dec0197093e4/lib/modules/manager/npm/extract/yarn.ts#L102-L116

Reply:
so simply return lock constraint?